app/controller.go

```diff
@@ -108,9 +109,15 @@ func startController(c *cli.Context) error {
 		frontend = f
 	}
 
-	control := controller.NewController(name, frontendIP, clusterIP, dynamic.New(factories), frontend, int(replicationFactor))
+	startAutoSnapDeletion := make(chan bool)
+	control := controller.NewController(name, frontendIP, clusterIP, dynamic.New(factories), frontend, int(replicationFactor), startAutoSnapDeletion)
```

Comment:
arguments are increasing.. doesn't look correct

Reply:
Think of using controller options